nodejs/eks/cluster.ts

```diff
@@ -120,11 +141,22 @@ function createOrGetInstanceProfile(name: string, parent: pulumi.ComponentResour
     return instanceProfile;
 }
 
-function generateKubeconfig(clusterName: string, clusterEndpoint: string, certData: string, roleArn?: string) {
+function generateKubeconfig(
+    clusterName: pulumi.Input<string>,
+    clusterEndpoint: pulumi.Input<string>,
+    certData: pulumi.Input<string>,
+    opts?: KubeconfigOptions) {
     let args = ["token", "-i", clusterName];
+    let env: { [key: string]: pulumi.Input<string>} | undefined;
 
-    if (roleArn) {
-        args = [...args, "-r", roleArn];
+    if (opts?.roleArn) {
+        args = [...args, "-r", opts.roleArn];
+    }
+    if (opts?.profileName) {
```

Comment:
Should these options be mutually exclusive?

Reply:
No, I thought the same initially.

Per the [AWS docs](https://docs.aws.amazon.com/eks/latest/userguide/create-kubeconfig.html) they can be used independently, or additively.

- The `roleArn` is an arg in the authentication, and
- The `profileName` is an environment setting

e.g. A few scenarios:
- `roleArn`: Assume this role on the default account caller
- `profileName`: Use this creds profile instead of the default account caller
- `roleArn` + `profileName`: Use this creds profile instead of the default account caller, and assume a role on the profile. This step is also possible with only a `profileName`, iff the profile already includes a role to assume in its settings.

Thanks for pointing this out. I'll add a note to the docs & OP to make this clear